src/vs/platform/extensionManagement/node/positronBootstrapExtensionsInitializer.ts

```diff
@@ -0,0 +1,114 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2025 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { IExtensionManagementService } from '../common/extensionManagement.js';
+import { IExtensionManifest } from '../../extensions/common/extensions.js';
+import { URI } from '../../../base/common/uri.js';
+import { join } from 'path';
+import { ILogService } from '../../log/common/log.js';
+import { IProductService } from '../../product/common/productService.js';
+import { INativeEnvironmentService } from '../../environment/common/environment.js';
+import { FileOperationResult, IFileService, IFileStat, toFileOperationResult } from '../../files/common/files.js';
+import { getErrorMessage } from '../../../base/common/errors.js';
+import { existsSync, readFileSync, writeFileSync } from 'fs';
+import { Disposable } from '../../../base/common/lifecycle.js';
+
+export class PositronBootstrapExtensionsInitializer extends Disposable {
+	private readonly storageFilePath: string;
+
+	constructor(
+		@INativeEnvironmentService private readonly environmentService: INativeEnvironmentService,
+		@IExtensionManagementService private readonly extensionManagementService: IExtensionManagementService,
+		@IFileService private readonly fileService: IFileService,
+		@IProductService private readonly productService: IProductService,
+		@ILogService private readonly logService: ILogService
+	) {
+		super();
+
+		this.storageFilePath = join(this.getVSIXPath().fsPath, '.version');
+		const currentVersion = this.productService.positronVersion;
+		const lastKnownVersion = existsSync(this.storageFilePath) ? readFileSync(this.storageFilePath, 'utf8').trim() : '';
```

Comment:
Can `IStorageService` be used here to retrieve and store the last known version?

Reply:
`IStorageService` is only available in the desktop code. I believe this is why Microsoft's feature is written only as a desktop utility. It felt cleaner to keep the desktop and server code all together rather than having duplicated logic per implementation, but the cost is maintaining the `.version` file.